server/index.js

```diff
@@ -52,6 +52,97 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (req.session.cartId) {
+    const sql = `
+      select "c"."cartItemId",
+              "c"."price",
+              "p"."productId",
+              "p"."image",
+              "p"."name",
+              "p"."shortDescription"
+          from "cartItems" as "c"
+          join "products" as "p" using ("productId")
+        where "c"."cartId" = $1
+    `;
+    db.query(sql, [req.session.cartId])
+      .then(result => {
+        res.json(result.rows[0]);
+      })
+      .catch(err => next(err));
+  } else {
+    res.json([]);
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const { productId } = req.body;
+  const productIdInt = parseInt(productId);
+  if (!Number.isInteger(productIdInt) || productIdInt <= 0) {
+    next(new ClientError('productId must be a positive number', 400));
+  }
+  const sql = `
+  select "price",
+      "productId"
+    from "products"
+    where "productId" = $1
+  `;
+  db.query(sql, [productId])
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`Cannot find product with "productId" ${productId}`, 404));
```

Comment:
Since we're in a promise chain, you'll need to actually `throw new ClientError(message, statusCode)` to break out to the `.catch()`, otherwise, the rest of the code will continue to run.